### PR TITLE
Patch/sgrouphydrogens

### PR DESCRIPTION
**Pull request by johnmay**
From discussion https://github.com/cdk/depict/issues/7... refine hydrogen suppression context.

### CODE REVIEW

File: base/core/src/main/java/org/openscience/cdk/sgroup/Sgroup.java [hunk 1]
```diff
@@ -114,14 +114,22 @@ public final SgroupType getType() {
     }
 
     /**
-     * Add a bond to this Sgroup.
+     * Add an atom to this Sgroup.
```
Comment (egonw) [inline base/core/src/main/java/org/openscience/cdk/sgroup/Sgroup.java]: Wow... so much for peer review.

File: base/core/src/main/java/org/openscience/cdk/tools/manipulator/SgroupManipulator.java [hunk 2]
```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright (c) 2018 John Mayfield
```
Comment (egonw) [inline base/core/src/main/java/org/openscience/cdk/tools/manipulator/SgroupManipulator.java]: Ah, right, the yearly README update :)

File: base/core/src/main/java/org/openscience/cdk/tools/manipulator/SgroupManipulator.java [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+        if (map == null)
+            return obj;
+        T val = (T) map.get(obj);
+        if (val == null)
+            return obj;
+        return val;
+    }
+
+    /**
+     * Copy a collection of Sgroups, replacing any IAtom/IBond (ChemObject)
```
Comment (egonw) [inline base/core/src/main/java/org/openscience/cdk/tools/manipulator/SgroupManipulator.java]: Option to use {@link}...

**Comment by johnmay** [comment]
Added dedicated tests

**Pull request merged**